### ISSUE TITLE
Deprecation warnings since Home Assistant 2022.6.0

### ISSUE THREAD

**Issue by megapearl** [jpcornil-git/HA-mcp23017#9]
Hi,

After updating to Home Assistant 2022.6.0 the component gives some deprecation warnings:

```
Logger: homeassistant.helpers.frame
Source: helpers/frame.py:103
First occurred: 10:47:43 (1 occurrences)
Last logged: 10:47:43

Detected integration that uses deprecated `async_get_registry` to access device registry, use async_get instead. Please report issue to the custom component author for mcp23017 using this method at custom_components/mcp23017/__init__.py, line 150: devices = await device_registry.async_get_registry(hass)
```

I tried to change 'devices = await device_registry.async_get_registry(hass)' to 'devices = await device_registry.async_get(hass)' at line 150 in __init__.py as mentioned above, but that gives the error below, so it's not as simple as that unfortunately.

```
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.9/site-packages/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/home/homeassistant/.homeassistant/custom_components/mcp23017/binary_sensor.py", line 72, in async_setup_entry
    binary_sensor_entity.device = await async_get_or_create(
  File "/home/homeassistant/.homeassistant/custom_components/mcp23017/__init__.py", line 150, in async_get_or_create
    devices = await device_registry.async_get(hass)
TypeError: object DeviceRegistry can't be used in 'await' expression
```

The component though is working fine, so no rush, but if you have some time maybe you can look into it.

Best Regards,
Donald..

**Comment by megapearl** [comment]
Tested Commit  #66cb6d1 (using HACS 'main' branch) and this does not give any deprecation warnings anymore.
Thanks!

**Comment by jpcornil-git** [comment]
This is now included in v1.1.1 (bumped homeassistant requirement to 2021.3.0 to match related change)
